docs/Host_Validation_Architecture.md

```diff
@@ -0,0 +1,25 @@
+# Host Validation and Testing
+
+__Date:__  7/6/2017
+
+__Objective:__ To determine the best strategy to validate hosts have joined the Kubernetes cluster successfully and are ready to schedule workloads
+
+## Problem Statement
+The Promenade project is a simple and efficient tool for orchestrating the deployment of self-hosted Kubernetes clusters.  However, Promenade currently lacks the ability to validate that clusters have been deployed successfully and are actually ready to schedule workloads.   I would like to propose a series of tests and validations that can be included with Promenade which will ensure the Kubernetes clusters are working as intended at the time of deployment, and can be executed at any point in time  after the cluster has been deployed to ensure it is continuing to function as intended for Day 2 operations. This tool should be able to identify potential problems with the Kubernetes cluster and provide the user with a comprehensive debugging output which highlights the identified issues.
+
+## Identified Tests
+I have identified the following tests which I believe to be a good base-line for basic Kubernetes smoke testing at the time of cluster deployment and during cluster life-cycle management:
+
+1. __Generate SHA256 hashes of all Promenade Generated configuration files and certificates:__ This process will run during the initial deployment of all the Kubernetes hosts in the cluster. Promenade will write a file to `/var/log/promenade/promenade_file_hash.log` with the path to the file as well as the SHA256 hash of that file when it was originally generated. Subsequent runs of the host validation script will re-calculate the SHA256 hashes of the files as they are presently, and compare them with the values written in the log file at the time of deployment. If the file hashes differ, the host validation script will log a failure and provide the user with a list of files that have potentially been modified.
+
+2. __Issue a REST Post Request to the Kubernetes API:__ This test will trigger a simple POST request to the internernal and external Kubernetes API and validate that it gets a `200` response.  This test will check to see if the Kubernetes API is running and responding to requests as needed.
+
+3. __Kubelet and Docker Validation__ : The goal of this test will be to ensure the Kubelet and Docker are present on the host and running without error.
+
```

Comment:
Good idea.  I completely agree.  That would give us great insight into the overall cluster health --- however, I feel as though more complicated tests like that are better suited for purpose-built monitoring tools (feel free to disagree).     I think the purpose of this document is to provide clear scoping as to what the expectations of Promenade are vs these other tools so we aren't re-inventing the wheel as such.    I can be convinced this would fall into the scope of Promenade, but I think we need to be super clear where the boundaries are.   I think that Promenade should do simple smoke testing to ensure cluster functionality, but would we want to evolve Promenade into a fully functional monitoring and config management platform? Or at that point, is it better to leverage other tools?  The questions I am posing in this document are what should those tests and configuration management tasks should be from the perspective of Promenade.